### ISSUE TITLE
🐛 Panic `attempt to multiply with overflow` in `no_useless_backref_in_regex` rule

### ISSUE THREAD

**Issue by qarmin** [biomejs/biome#5989]
### Environment information

Yesterday version, self compiled - 65e126793c122c3161991aec57b39104d27fb00b

Ubuntu 24.10

### What happened?

File content(at the bottom should be attached raw, not formatted file - github removes some non-printable characters, so copying from here may not work):
```
/\217483
```

command
```
timeout -v 150 biome lint TEST___FILE.js
```

App was compiled with nightly rust compiler to be able to use address sanitizer
(You can ignore this part if there is no address sanitizer error)
On Ubuntu 24.04, the commands to compile were:
```
rustup default nightly
rustup component add rust-src --toolchain nightly-x86_64-unknown-linux-gnu
rustup component add llvm-tools-preview --toolchain nightly-x86_64-unknown-linux-gnu

export RUST_BACKTRACE=1 # or full depending on project
export ASAN_SYMBOLIZER_PATH=$(which llvm-symbolizer-18)
export ASAN_OPTIONS=symbolize=1
RUSTFLAGS="-Zsanitizer=address" cargo +nightly build --target x86_64-unknown-linux-gnu
```

cause this
```
Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue.

When opening the issue, please provide a minimal reproduction, or identify and share the file/code that triggers it. Without a way to reproduce the error, the error can't be fixed:

Source Location: crates/biome_js_analyze/src/lint/nursery/no_useless_backref_in_regex.rs:327:46
Thread Name: biome::workspace_worker_0
Message: attempt to multiply with overflow


##### Automatic Fuzzer note, output status "None", output signal "Some(6)"
```

[compressed.zip](https://github.com/user-attachments/files/20166331/compressed.zip)

### Expected result

No panic overflow

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by Conaclos** [comment]
However, I think I can harden the rule. We should not overflow in cases like `//\217483/
